lib/AnyEvent/Handle/UDP.pm

```diff
@@ -230,11 +232,11 @@ sub _on_addr {
 
 	AnyEvent::Socket::resolve_sockaddr($host, $port, 'udp', $get_family->($self, $fh), Socket::SOCK_DGRAM, sub {
 		my @targets = @_;
-		while (1) {
-			my $target = shift @targets or $self->_error(1, "Could not resolve $host:$port");
-			$on_success->(@{$target});
-			last;
+		while (@targets) {
+			my $target = shift @targets;
```

Comment:
is here a `or next` missing?

The previous code throw an error if @targets == 0 or $target == false, the new code only if @targets == 0

Reply:
I don't think `$target` can ever be false (it only was in the old code because shifting an empty array yields an undef)